docs/writing-stories/mocking-modules.md

```diff
@@ -0,0 +1,229 @@
+---
+title: Mocking modules
+---
+
+Components can also depend on modules that are imported into the component file. These can be from external packages or internal to your project. When rendering those components in Storybook or testing them, you may want to mock those modules to control their behavior.
+
+There are two primary approaches to mocking modules in Storybook. They both involve creating a mock file to replace the original module. The difference between the two approaches is how you import the mock file into your component.
+
+For either approach, relative imports of the mocked module are not supported.
+
+## Mock files
+
+To mock a module, create a file with the same name and in the same directory as the module you want to mock. For example, to mock a module named `session`, create a file next to it named `session.mock.js|ts`, with a few characteristics:
+
+- It should re-export all exports from the original module.
+- It should use the `fn` utility to mock any necessary functionality from the original module.
+- It cannot use subpath imports or builder aliases for its dependencies.
+- It should not introduce side effects that could affect other tests or components. Mock files should be isolated and only affect the module they are mocking.
+
+Here's an example of a mock file for a module named `session`:
+
+```js
+// session.mock.js
+import { fn } from '@storybook/test';
+import * as actual from './session';
+
+export * from './session';
+export const getUserFromSession = fn(actual.getUserFromSession);
+```
+
+## Subpath imports
+
+The recommended method for mocking modules is to use [subpath imports](https://nodejs.org/api/packages.html#subpath-imports), a feature of Node packages that is supported by both [Vite](../builders/vite.md) and [Webpack](../builders/webpack.md).
+
+To configure subpath imports, you define the `imports` property in your project's `package.json` file. This property maps the subpath to the actual file path. The example below configures subpath imports for the `api` and `prisma` internal modules:
+
+TK: External module example?
```

Comment:
@JReinhold — Want to take a first crack at that?

Reply:
Added some words and an example. I'm not sure the example is necessary as it's very similar to the one above. I also don't fully love it because users today rarely use `'uuid'` but instead `crypto.randomUUID()`, but that wouldn't really be an example of mocking an external module..